cartographer/io/serialization_format_migration.cc

```diff
@@ -18,284 +18,200 @@
-        legacy_data.trajectory_data();
-  }
-  if (legacy_data.has_imu_data()) {
-    LOG_FIRST_N(INFO, 1) << "Migrating IMU data.";
-    auto& output_vector = (*proto_map)[SerializedData::kImuDataFieldNumber];
-    output_vector.emplace_back();
-    *output_vector.back().mutable_imu_data() = legacy_data.imu_data();
-  }
-  if (legacy_data.has_odometry_data()) {
-    LOG_FIRST_N(INFO, 1) << "Migrating odometry data.";
-    auto& output_vector = (*proto_map)[SerializedData::kOdometryData];
-    output_vector.emplace_back();
-    *output_vector.back().mutable_odometry_data() = legacy_data.odometry_data();
-  }
-  if (legacy_data.has_fixed_frame_pose_data()) {
-    LOG_FIRST_N(INFO, 1) << "Migrating fixed frame pose data.";
-    auto& output_vector =
-        (*proto_map)[SerializedData::kFixedFramePoseDataFieldNumber];
-    output_vector.emplace_back();
-    *output_vector.back().mutable_fixed_frame_pose_data() =
-        legacy_data.fixed_frame_pose_data();
-  }
-  if (legacy_data.has_landmark_data()) {
-    LOG_FIRST_N(INFO, 1) << "Migrating landmark data.";
-    auto& output_vector =
-        (*proto_map)[SerializedData::kLandmarkDataFieldNumber];
-    output_vector.emplace_back();
-    *output_vector.back().mutable_landmark_data() = legacy_data.landmark_data();
-  }
-  return true;
-}
 
-ProtoMap ParseLegacyData(
-    cartographer::io::ProtoStreamReaderInterface* const input,
-    bool migrate_grid_format) {
-  ProtoMap proto_map;
-  CHECK(ReadPoseGraph(input, &proto_map))
-      << "Input stream seems to differ from original stream format. Could "
-         "not "
-         "read PoseGraph as first message.";
-  CHECK(ReadBuilderOptions(input, &proto_map))
-      << "Input stream seems to differ from original stream format. Could "
-         "not "
-         "read AllTrajectoryBuilderOptions as second message.";
-  if (migrate_grid_format) {
-    do {
-    } while (DeserializeNextAndMigrateGridFormat(input, &proto_map));
-  } else {
-    do {
-    } while (DeserializeNext(input, &proto_map));
+  // Set global poses of landmarks.
+  for (const auto& landmark : pose_graph_proto.landmark_poses()) {
+    pose_graph.SetLandmarkPose(landmark.landmark_id(),
+                               transform::ToRigid3(landmark.global_pose()),
+                               true);
   }
-  return proto_map;
-}
 
-mapping::proto::SerializationHeader CreateSerializationHeader() {
-  constexpr uint32_t kVersion1 = 1;
-  mapping::proto::SerializationHeader header;
-  header.set_format_version(kVersion1);
-  return header;
-}
+  mapping::MapById<mapping::SubmapId, mapping::proto::Submap>
+      submap_id_to_submap;
+  mapping::MapById<mapping::NodeId, mapping::proto::Node> node_id_to_node;
+  SerializedData proto;
+  while (deserializer.ReadNextSerializedData(&proto)) {
+    switch (proto.data_case()) {
+      case SerializedData::kPoseGraph:
+        LOG(ERROR) << "Found multiple serialized `PoseGraph`. Serialized "
```

Comment:
Could we even be more defensive and `LOG(FATAL)`?

Reply:
Done.